### DISCUSSION TITLE
using FontCollection and TypefaceFontProvider

### DISCUSSION THREAD

**Discussion by Eywek** [rust-skia/rust-skia#781] [Q&A] [answered]
Hello,

I'm trying to fonts implementations coming from `skia_safe::textlayout` (to draw some SkParagraph later) and I'm hitting a wall by trying to configure the FontCollection:

```rust
fn main() {
    let font_provider = skia_safe::textlayout::TypefaceFontProvider::new();
    let mut font_collection = skia_safe::textlayout::FontCollection::new();

    font_collection.set_default_font_manager(Some(font_provider.into()), None);

    let typeface = skia_safe::Typeface::from_data(skia_safe::Data::new_empty(), 0).unwrap();
    font_provider.register_typeface(typeface, None::<String>); // err: borrow of moved value: `font_provider`
}
```

I need to init a `TypefaceFontProvider` and a `FontCollection` where the `FontCollection` would use the `TypefaceFontProvider` as `FontMgr`. And then later, I would load some fonts into my `TypefaceFontProvider`, but I'm unable to do it since Rust think the `TypefaceFontProvider` as been moved.

But from what I've seen (and experienced in c++), the `FontCollection` only use a pointer to the `TypefaceFontProvider`, so why can't we provide a pointer to the `set_default_font_manager` method instead of moving it? 

Thank you

**Comment by pragmatrix** [comment, marked as answer]
Because it's a reference counted type, you should be able to `.clone()` the `TypeFaceFontProvider` or the underlying `FontMgr`.
